web/profiles/custom/os2loop/themes/os2loop_theme/templates/content/node--os2loop-post--list-display.html.twig

```diff
@@ -70,5 +70,6 @@
  *   in different view modes.
  */
 #}
-{% set body_text = node.os2loop_post_content.value|striptags|raw %}
+{# // Remove whitespace added when user copy/pastes content in form. #}
+{% set body_text = node.os2loop_post_content.value|replace({'&nbsp;': ' '})|striptags|raw %}
```

Comment:
This doesn't look right to me (and does not handle other exotic characters, like `<` and `&`, that a user may input).

I think we need a Twig filter for stripping and truncating text, i.e. something like https://git.drupalcode.org/project/twig_truncate/-/blob/8.x-1.x/src/TwigExtension/Truncate.php (https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Unicode.php/function/Unicode%3A%3Atruncate/9.2.x) plus use of https://www.php.net/manual/en/function.html-entity-decode.php and https://www.php.net/manual/en/function.strip-tags.php to clean up before truncating.

Can't Drupal do this for us?

What we really need is https://www.drupal.org/project/no_nbsp or similar stuff to clean up text that a rouge user enters or copies from some dangerous source.

Reply:
I chose a different approach and no longer use the db value but rather the rendered content